docs/source/en/optimization/attention_backends.md

```diff
@@ -81,6 +81,44 @@ with attention_backend("_flash_3_hub"):
 > [!TIP]
 > Most attention backends support `torch.compile` without graph breaks and can be used to further speed up inference.
 
+## Checks
+
+The attention dispatcher includes debugging checks that catch common errors before they cause problems.
+
+1. Device checks verify that query, key, and value tensors live on the same device.
+2. Data type checks confirm tensors have matching dtypes and use either bfloat16 or float16.
+3. Shape checks validate tensor dimensions and prevent mixing attention masks with causal flags.
+
+Enable these checks by setting the `DIFFUSERS_ATTN_CHECKS` environment variable.
+
+```py
+import os
+import torch
+
+os.environ["DIFFUSERS_ATTN_CHECKS"] = "1"
+
+query = torch.randn(1, 10, 8, 64, dtype=torch.bfloat16, device="cuda")
+key = torch.randn(1, 10, 8, 64, dtype=torch.bfloat16, device="cuda")
+value = torch.randn(1, 10, 8, 64, dtype=torch.bfloat16, device="cuda")
+
+try:
+    with attention_backend("flash"):
+        output = dispatch_attention_fn(query, key, value)
+        print("✓ Flash Attention works with checks enabled")
+except Exception as e:
+    print(f"✗ Flash Attention failed: {e}")
```

Comment:
I think this needs to be set as `export DIFFUSERS_ATTN_CHECKS=yes` before running any execution because of how `DIFFUSERS_ATTN_CHECKS` is used:
https://github.com/huggingface/diffusers/blob/b4e6dc3037e75e2dc16466b914b1be597b06be9a/src/diffusers/models/attention_dispatch.py#L43

https://github.com/huggingface/diffusers/blob/b4e6dc3037e75e2dc16466b914b1be597b06be9a/src/diffusers/utils/constants.py#L45

Reply:
Yeah `DIFFUSERS_ATTN_CHECKS` is determined at the module-level so I am not sure setting it that way would be effective.